lib/sparse/src/common/sparse_vector.rs

```diff
@@ -87,6 +87,7 @@ impl SparseVector {
     }
 
     /// Construct a new vector that is the result of performing all indices-wise operations
+    /// Warning: Expects both vectors to be sorted by indices.
```

Comment:
we can check while iteration that indices are growing

Reply:
I find that there are debug_assertion for that. @agourlay do you know why the debug assertions wasn't triggered?

Reply:
the assertion is triggered on a debug build for the Python congruence tests :)